src/auth_db/mod.rs

```diff
@@ -8,23 +8,73 @@ use std::{
 
 use hex;
 use reqwest::{Client as RequestClient, Error as RequestError, StatusCode, Url, UrlError};
+use serde::de::{Deserialize, Deserializer, Error as DeserializeError, Unexpected};
 
 use settings::Settings;
 
 #[cfg(test)]
 mod test;
 
-#[derive(Clone, Copy, Debug, Deserialize, Eq, Hash, PartialEq)]
+#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq)]
 pub enum BounceType {
-    Hard = 1,
-    Soft = 2,
-    Complaint = 3,
+    Hard,
+    Soft,
+    Complaint,
+}
+
+impl<'d> Deserialize<'d> for BounceType {
+    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
+    where
+        D: Deserializer<'d>,
+    {
+        let value: u8 = Deserialize::deserialize(deserializer)?;
+        match value {
+            0 => {
+                // Set whenever the auth db receives a value it doesn't recognise
+                println!("Mapped default auth db bounce type BounceType::Soft");
+                Ok(BounceType::Soft)
+            }
+            1 => Ok(BounceType::Hard),
+            2 => Ok(BounceType::Soft),
+            3 => Ok(BounceType::Complaint),
+            _ => Err(D::Error::invalid_value(
+                Unexpected::Unsigned(value as u64),
+                &"bounce type",
+            )),
+        }
+    }
 }
 
 #[derive(Clone, Copy, Debug, Deserialize)]
+pub enum BounceSubtype {
+    // Set by the auth db if an input string is not recognised
+    Unmapped = 0,
+    // These are mapped from the equivalent SES BounceSubType values
+    Undetermined = 1,
+    General = 2,
+    NoEmail = 3,
+    Suppressed = 4,
+    MailboxFull = 5,
+    MessageTooLarge = 6,
+    ContentRejected = 7,
+    AttachmentRejected = 8,
+    // Not sure where these come from
```

Comment:
Found some information one these [here](https://docs.aws.amazon.com/ses/latest/DeveloperGuide/notification-contents.html#bounce-types).

Reply:
Sorry, probably wasn't clear, this comment is about the values that come after it. They're not from SES (at least they're not mentioned in that doc from what I can see), but they are in the db server and I don't understand why.

Reply:
Oh, my bad, they are there as the complaint types. I was looking for them in a table like the bounce type stuff, should have searched. 😊

I'll get rid of the comment.